prusti-viper/src/encoder/encoder.rs

```diff
@@ -1280,18 +1288,18 @@ impl<'v, 'tcx> Encoder<'v, 'tcx> {
         proc_def_id: ProcedureDefId,
         arg_ty: ty::Ty<'tcx>, // type arguments
         is_equality: bool // true = equality, false = disequality
-    ) -> (String, vir::Type) {
+    ) -> Result<(String, vir::Type), EncodingError> {
         let snapshot_res = self.encode_snapshot(&arg_ty);
```

Comment:
```suggestion
        let snapshot_res = self.encode_snapshot(&arg_ty)?;
```

And remove the `is_ok` check etc?

Reply:
It's not very readable, but the point is that encoding errors of the snapshot should be catched and converted to stubs, while errors raising from the encoding of the stub should not be catched.

Reply:
Ah ok. Without the additional `is_defined` check I would suggest this be a `.map_or_else` operation. I wonder what it means to successfully encode a snapshot that is not "defined". @cmatheja ?

Reply:
I wonder the same. I think that there is some overlapping between returning a "not defined" snapshot and a snapshot encoding error. I was the one who added `EncodingError`s to the snapshot encoder to fix some panics, and I did so trying not to change the behavior of the encoder. Now that the encoding errors are usable, we could get rid of the `is_defined()` technique, but we should check with Christoph to be sure.